Sources/SwiftFrameCore/Workers/ImageComposer.swift

```diff
@@ -0,0 +1,106 @@
+import AppKit
+import Foundation
+import CoreGraphics
+import CoreImage
+
+final class ImageComposer {
+
+    // MARK: - Properties
+
+    private let textRenderer = TextRenderer()
+    private let screenshotRenderer = ScreenshotRenderer()
+
+    let context: CGContext
+
+    // MARK: - Init
+
+    init(canvasSize: CGSize) throws {
+        self.context = try ImageComposer.createContext(size: canvasSize)
+    }
+
+    // MARK: - Preparation
+
+    private static func createContext(size: CGSize) throws -> CGContext {
+        guard let context = CGContext(
+            data: nil,
+            width: Int(size.width),
+            height: Int(size.height),
+            bitsPerComponent: 8,
+            bytesPerRow: 0,
+            space: CGColorSpaceCreateDeviceRGB(),
+            bitmapInfo: CGImageAlphaInfo.premultipliedLast.rawValue)
+        else {
+            throw NSError(description: "Failed to create graphics context")
+        }
+        return context
+    }
+
+    // MARK: - Composition
+
+    func addTemplateImage(_ image: NSBitmapImageRep) throws {
+        guard let templateImage = image.cgImage else {
+            throw NSError(description: "Could not render template image")
+        }
+
+        context.saveGState()
+        defer { context.restoreGState() }
```

Comment:
Nit: Since this method only has one drawing command `defer` almost seems like overkill. Just doing

```
save()
draw()
restore()
```

_might_ actually be easier to understand.

Reply:
Definitely, especially for someone who's unfamiliar with `defer`